cloud/satellite/satellite_activation_keys.py

```diff
@@ -0,0 +1,365 @@
+    - This module provides Satellite activation key management via the REST API for Red Hat Satellite system management servers.
+options:
+    user:
+        description:
+            - The name of the Satellite user.
+        required: true
+    pwd:
+        description:
+            - The password assigned to the username.
+        required: true
+    satellite:
+        description:
+            - The host name of Satellite.
+        required: true
+    organization:
+        description:
+            - The organization identifier used in Satellite
+        required: true
+    key_action:
+        description:
+            - The actual action performed on a key.
+        required: true
+        choices: ['create','update','delete']
+    key_name:
+        description:
+            - The name of the managed key.
+        required: true
+    validate_certs:
+        description:
+            - If SSL vertificate errors should be ignored or not.
+        default: true
+        required: false
+    key_description:
+        description:
+            - Description of the purpose for the key.
+        default: null
+        required: false
+    environment:
+        description:
+            - The environment of the activation key.
+        default: null
+        required: false
+    content_view:
+        description:
+            - The content view of the activation key.
+        default: null
+        required: false
+    max_content_hosts:
+        description:
+            - The maximum number of registered hosts.
+        default: null
+        required: false
+    unlimited_content_hosts:
+        description:
+            - Describe if the activation key can have unlimited hosts.
+        default: null
+        required: false
+    release_version:
+        description:
+            - Content release version of the authorization key.
+        default: null
+        required: false
+    service_level:
+        description:
+            - Service level of the authorization key.
+        default: null
+        required: false
+    auto_attach:
+        description:
+            - Auto attach subscription upon registration.
+        default: null
+        required: false
+'''
+
+EXAMPLES = '''
+Example playbook entries using the satellite_activation_keys module to manage Satellite activation keys.
+
+    tasks:
+
+    - name: create new key
+      satellite_activation_keys:
+        user: admin
+        pwd: secret
+        satellite: "satellite.example.com"
+        key_action: create
+        key_name: mynewkey
+        organization: my_organization
+'''
+
+RETURN = '''
+msg:
+    description: status response of the REST call
+    returned: success
+    type: string
+    sample: {\"organization\":{\"name\":\"myorg\"}}
+'''
+
+def get_content_view_id(module,orga_id):
+    """Looks up content view ID for a given content view name.
+    """
+
+    url_action = 'https://' + module.params.get('satellite') + '/katello/api/v2/content_views/'
```

Comment:
it seems this could safely be moved out to a constant as it stands this is a magic string